APA PsycNET.js

```diff
@@ -1,162 +1,199 @@
+		if (!href || !title) continue;
+		if (checkOnly) return true;
+		found = true;
+		items[href] = title;
 	}
+	return found ? items : false;
 }
 
-//default field xpath
-var fields = {
-	title: '(//div[@id="rdcTitle"])[1]',
-	authors: '(//div[@id="rdcAuthors"])[1]',
-	voliss: '(//div[@id="rdcSource"])[1]',
-	abstract: '//div[@id="rdRecord"]/div[@class="rdRecordSection"][2]'
-}
 
-function getField(field, doc) {
-	var val = ZU.xpathText(doc, field);
-	if(val) val = ZU.trimInternal(val);
-	return val;
+function doWeb(doc, url) {
+	if (detectWeb(doc, url) == "multiple") {
+		Zotero.selectItems(getSearchResults(doc, false), function (items) {
+			if (!items) {
+				return true;
+			}
+			var articles = [];
+			for (var i in items) {
+				articles.push(i);
+			}
+			ZU.processDocuments(articles, scrape);
+		});
+	} else {
+		scrape(doc, url);
+	}
 }
 
-//for scraping publication information directly from pages
-var volissRe = {
-	journalArticle: 
-		/^(.+?)(?:,\sVol\s(\d+)\((\d+)\))?,\s(\w+\s(?:\d+\s*,\s)?\d{4}),\s(?:(\d+-\d+)|No Pagination Specified).(?:\sdoi:\s(.+))?$/i,
-	bookSection:
-		/^(.+?),\s\((\d{4})\)\.\s(.+?),\s\(pp\.\s(\d+-\d+)\)\.\s(.+?):\s(.+?),\s(?:(\w+))?,\s(\d+)\spp\.(?:\sdoi:\s(.+))?/i,
-	book:
-		/^(.+?):\s(.+?)(?:\.\s\((\d{4})\)\.\s(\w+)\s(\d+)\spp\.\sdoi:\s(.+))?$/i
-};
 
-var creatorMap = {
-	Ed: 'editor'
-};
-
-function doWeb(doc, url) {
-	var type = detectWeb(doc, url);
-	if (type == "multiple") {
-		var items = new Object();
-		var titles = ZU.xpath(doc, '//div[@class="srhcTitle"]/a');
-		for(var i=0, n=titles.length; i<n; i++) {
-			items[titles[i].href] = titles[i].textContent;
+function scrape(doc, url) {
+	var uid = getIds(doc,url);
+	
+	var productCode;
+	var db = doc.getElementById('database');
+	if (db) {
+		var db = db.parentNode.textContent;
+		if (db.indexOf('PsycARTICLES')>-1) {
+			productCode = 'PA';
+		} else if (db.indexOf('PsycBOOKS')>-1) {
+			productCode = 'PB';
+		} else if (db.indexOf('PsycINFO')>-1) {
+			productCode = 'PI';
 		}
-		Zotero.selectItems(items, function(selectedItems) {
-			if(!selectedItems) return true;
+	}
+	Z.debug(productCode);
 	
-			var arts = new Array();
-			for(var i in selectedItems) {
-				arts.push(i);
-			}
-			ZU.processDocuments(arts, scrape);
+	var postData = '{"api":"record.exportRISFile","params":{"UIDList":[{"UID":"'+uid+'","ProductCode":"'+productCode+'"}],"exportType":"zotero"}}';
+	var headers = {
+		'Content-Type': 'application/json',
+		'Referer': url
+	};
+	Z.debug(postData);
+
+
+	ZU.doPost('/api/request/record.exportRISFile', postData, function(text) {
+		try {
+			var data = JSON.parse(text);
+		} catch(e) {
+			Z.debug('POST request did not result in valid JSON');
+			Z.debug(text);
+		};
+Z.debug(text)
+		if (data && data.isRisExportCreated) {
+			Z.debug(data);
+			ZU.doGet('/ris/download', function(text2) {
+				Z.debug("TEXT 2: "+text2);
+				var redirect = text2.match(/<meta http-equiv="refresh" content="10; url=([^"]*)"/)
```

Comment:
I have experienced this, but after solving this once, it didn't occur any more. Maybe this occurs the first time you are opening their website? That would be good to test more...